visualizer/src/topology.js

```diff
@@ -196,18 +79,174 @@ module.exports = class Topology {
+                info.innerHTML = `
+                    <div class='portinfo'>${portName}</div>
+                    <div class='portinfo'>Bandwidth: ${portInfo.write_port.bandwidth}</div>
+                    <div class='portinfo'>Latency: ${portInfo.read_ports.latency}</div>`;
+            });
+            c.bind('mouseout', () => {
+                const info = document.querySelector('#infobox');
+                info.style.animation = 'infobox_disappear 0.5s ease-in-out 0s 1 normal forwards';
+            })
+        }
+    }
+
+    configureConnections() {
+        const connectionTypeMap = this.createConnectionTypeMap();
+        for (const [portName, portInfo] of Object.entries(this.data.portmap)) {
+            for (const [moduleName, ports] of Object.entries(this.data.modules)) {
+                if (ports.write_ports !== '' && portName in ports.write_ports) {
+                    this.configureConnection(portName, portInfo, moduleName, connectionTypeMap);
+                }
+            }
+        }
+    }
+}
+
+class dagreConfig extends BaseConfig {
+    configureModules(modulemap, parent) {
+        for (const [moduleName, children] of Object.entries(modulemap)) {
+            this.graph.setNode(moduleName, { label: `${moduleName}_L`, width: this.moduleWidth, height: this.moduleHeight});
+            if (children !== "") {
+                this.configureModules(children, moduleName);
+            }
+            if (parent !== "") {
+                this.graph.setParent(moduleName, parent);
+            }
+        }
+    }
 
-    moduleConnectionNumber(mName) {
-        const connectionsNumber = {};
-        for (const moduleName of Object.keys(this.data.modules)) {
-            connectionsNumber[moduleName] = 0;
+    configureConnection(portName, moduleName) {
+        for (const targetName of this.modulesWithReadPort(portName, moduleName)) {
+            // Waiting fix : https://github.com/dagrejs/dagre/issues/236
+            const haveNoChildren = (graph, source, target) => {
+                let sourceC = Object.entries(graph._children[source]).length;
+                let targetC = Object.entries(graph._children[target]).length;
+                return sourceC === 0 && targetC === 0;
+            }
+            if (haveNoChildren(this.graph, moduleName, targetName)) {
+                this.graph.setEdge(moduleName, targetName);
+            }
         }
-        for (const portName of Object.keys(this.data.modules[mName])) {
-            for (const moduleName of this.modulesWithReadPort(portName, mName)) {
-                connectionsNumber[moduleName] += 1;
+    }
+
+    configureConnections() {
+        for (const portName of Object.keys(this.data.portmap)) {
+            for (const [moduleName, ports] of Object.entries(this.data.modules)) {
+                if (ports.write_ports !== '' && portName in ports.write_ports) {
+                    this.configureConnection(portName, moduleName);
+                }

```

Comment:
... and this code are 80% the same. Can it be moved to base class, so different classes would use different `configureConnection` methods?